### DISCUSSION TITLE
Weird color pixels after converting with a color palette

### DISCUSSION THREAD

**Discussion by dreamangel-1966** [ImageMagick/ImageMagick#2237] [Help]
I have a problem with an image "source.svg", that has a transparent background.
I extracted from this picture the most common colors.
These colors were saved in the color palette file "colormap.tif".
After converting the image "source.svg" with this color palette I presented orange color pixels
in the target file "target.tif" around all the letters.
Gimp tells me that these color pixels have the hex code "FF7F00".
According to the histogram, this color comes in the image "source.svg" most often before (see step 3).
How can i prevent it that these unwanted color pixels appear at the letters?

1. Download the picture "source.svg":

`wget "https://upload.wikimedia.org/wikipedia/commons/4/46/Aufbau_der_Erde_schematisch.svg" --output-document="source.svg"`

2. Determine the existence of the alpha channel:

`identify -verbose "source.svg" | grep 'Alpha:' | cut --delimiter=":" --fields="2" | sed -e 's/^\x20*//g;s/\x20\+/\x20/g;s/\x20*$//g'`

The result is "1-bit".
In this case the image "source.svg" has an alpha channel.

3. Determine the most common colors in the picture "source.svg":

`magick "source.svg" -format "%c" -depth "8" histogram:info: | sed -e 's/^\x20*//g;s/\x20\+/\x20/g;s/\x20*$//g' | sort --numeric-sort --reverse > "histogram.txt"`

According to the histogram "histogram.txt" the following colors come most often:

821572: (255,127,0,255) #FF7F00FF srgba(255,127,0,1)
587529: (255,255,255,255) #FFFFFFFF white
507133: (255,191,0,255) #FFBF00FF srgba(255,191,0,1)
274536: (255,63,0,255) #FF3F00FF srgba(255,63,0,1)
267287: (255,0,0,255) #FF0000FF red
72953: (255,255,0,255) #FFFF00FF yellow
53010: (0,0,0,255) #000000FF black

4. Create a color palette "colormap.tif" with the most common colors:

`magick +dither -size "1x1" xc:#FF7F00FF xc:#FFFFFFFF xc:#FFBF00FF xc:#FF3F00FF xc:#FF0000FF xc:#FFFF00FF xc:#000000FF +append -depth "8" "colormap.tif"`

The color palette "colormap.tif" also has an alpha channel:

`identify -verbose "colormap.tif" | grep 'Alpha:' | cut --delimiter=":" --fields="2" | sed -e 's/^\x20*//g;s/\x20\+/\x20/g;s/\x20*$//g'`

The result is "1-bit".

5. Convert the image "source.svg" with the color palette "colormap.tif":

`magick "source.svg" +dither -alpha "off" -depth "8" -remap "colormap.tif" "target.tif"`

Note that the alpha channel is removed during the conversion.

**Comment by snibgo** [comment]
The rasterized SVG has an alpha channel, but every pixel is fully opaque, so this makes no difference to the result.

The SVG has data that is specific to Inkscape, so it is best rasterized with Inkscape.

The letters are antialiased. This means the edges of the letters have colours between the letter colour (black) and the colour that surrounds the letter. "-remap" chooses, for each pixel, the closest colour in the map. At the anti-aliased edges of letters, this will sometimes be a colour that is neither black nor the surrounding colour. It seems you don't want this, which means you don't want anti-aliasing.

Inkscape has a GUI option for turning off Inkscape's anti-aliasing, but there seems to be no way in batch. But it can be done with MSVG:
```
magick +antialias MSVG:source.svg x.png
```

It is also possible to take an anti-aliased image, and guess which pixels are anti-aliased, make those transparent, and fill in the holes (aka inpainting). But the technique is messy, and the results may not be good.
